components/admin/app/controllers/tags_controller.rb

```diff
@@ -1,74 +1,80 @@
 class TagsController < ApplicationController
+  before_action :set_category
   before_action :set_tag, only: [:show, :edit, :update, :destroy]
 
-  # GET /tags
-  # GET /tags.json
+  # GET /category/:category_id/tags
+  # GET /category/:category_id/tags.json
```

Comment:
These are now strictly hierarchical under categories and as such should be accessed through them.